DSpace/dspace-angular
# Added input validation for new metadata field submissions & fixed some other small bugs

## References
* Fixes DSpace/DSpace#8753
* Fixes DSpace/DSpace#8754
* Related DSpace/DSpace#8758

## Description
In this PR I added the input validation to the Metadata Registry forms (and fixed some other small issues related to those forms) and I also set the max suggestion dropdown length back to 10. I also made the frontend changes to prevent users from editing the metadata schema Name, Element or qualifier by disabling those fields during the update.

## Instructions for Reviewers
List of changes in this PR:
* `MetadataFieldFormComponent` & `MetadataSchemaFormComponent`
  * Added validation for the fields to check that they don't contain any dots
  * Removed error from first field after successfully submitting the form or unselecting the field you are currently updating
  * For `MetadataSchemaFormComponent` only: cleared out the name field after successfully submitting the form
  * Disabled the schema Name, Element and Qualifier fields during the update process to prevent users from changing those values
* `MetadataFieldSelectorComponent`: The default page size on REST for all endpoints is 20 elements previously only 10 elements were shown because the pageable parameter on the rest side was not working. Now that it has been fixed (see related PR), I had to add that limit back so that only 10 suggestions are shown.

**Guidance for how to test & review this PR:**
- Go to `/admin/registries/metadata`
  - Try adding a valid new schema and check that both fields are cleared out & that the first field does not show an error
  - Fill in the Namespace input field with something containing a dot, check that an appropriate message is shown and that you can't submit it
- Go to `/admin/registries/metadata/dc`
  - Try adding a valid field and check that both fields are cleared out & that the first field does not show an error
  - Fill in the Element & Qualifier input fields with something containing a dot, check that an appropriate message is shown and that you can't submit it

## Checklist
